model/src/main/java/org/projectnessie/model/PartitionField.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright (C) 2020 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.model;
+
+import com.fasterxml.jackson.annotation.JsonTypeName;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import org.immutables.value.Value;
+
+@Value.Immutable
+@JsonSerialize(as = ImmutablePartitionField.class)
+@JsonDeserialize(as = ImmutablePartitionField.class)
+@JsonTypeName("ICEBERG_PARTITION_FIELD")
+public interface PartitionField {
+
+  int getSourceId();
+
+  int getFieldId();
+
+  String getName();
+
+  String getTransform();
```

Comment:
I think, this is an interface in Iceberg - not sure how that can be mapped to a string